tests/bsim/bluetooth/host/central/src/main.c

```diff
@@ -8,9 +8,14 @@
 #include "babblekit/testcase.h"
 #include <zephyr/bluetooth/conn.h>
 
+/* Include conn_internal for the purpose of checking reference counts. */
+#include <zephyr/../../subsys/bluetooth/host/conn_internal.h>
+
+struct bst_test_list *test_peripheral_install(struct bst_test_list *tests);
+
 static K_SEM_DEFINE(sem_connected, 0, 1);
 
-static void connected_cb(struct bt_conn *conn, uint8_t err)
+static void connected_cb_expect_fail(struct bt_conn *conn, uint8_t err)
```

Comment:
It's a bit odd to reuse `sem_connected` in this callback. I would suggest to add another semaphore/flag for this purpose, so that when you do 
```c
	err = k_sem_take(&sem_connected, K_FOREVER);
	TEST_ASSERT(err == 0, "Failed getting connected timeout", err);
```

in the test, it doesn't look like you are actually waiting for a connection, but for a connection failed event.